pkg/storage/watch/watch.go

```diff
@@ -0,0 +1,196 @@
+package watch
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/rjeczalik/notify"
+	log "github.com/sirupsen/logrus"
+	"github.com/weaveworks/ignite/pkg/storage"
+	"github.com/weaveworks/ignite/pkg/storage/watch/update"
+)
+
+const eventBuffer = 4096 // How many events and updates we can buffer before watching is interrupted
+var excludeDirs = []string{".git"}
+var listenEvents = []notify.Event{notify.InCreate, notify.InDelete, notify.InDeleteSelf, notify.InCloseWrite}
+
+var eventMap = map[notify.Event]update.Event{
+	notify.InCreate:     update.EventCreate,
+	notify.InDelete:     update.EventDelete,
+	notify.InCloseWrite: update.EventModify,
+}
+
+type eventStream chan notify.EventInfo
+type UpdateStream chan *update.FileUpdate
+type watches []string
+
+type watcher struct {
```

Comment:
directoryWatcher or inotifyWatcher would be a better name here IMO, I didn't understand what `watcher` did before looking at it

Reply:
It's only used internally (it's local), so I like to keep the name short, but I'll add a godoc comment to explain what it is and does.